doc/GUIDE.md

```diff
@@ -1712,8 +1717,43 @@ of a multi line block comment:
     --install-ghc
     runghc
     --package turtle
+    --
+    -hide-all-packages
   -}
 ```
+
+### Writing independent and reliable scripts
+
+Independent means that the script is independent of any prior deployment
+specific configuration. If required, the script will install everything it
+needs automatically on any machine that it runs on. To make a script always
+work irrespective of any specific environment configuration you can do the
+following:
+1. Use the `--install-ghc` option to install the compiler automatically
+2. Explicitly specify all packages required by the script using the
+`--package` option. Use `-hide-all-packages` ghc option to force
+explicit specification of all packages.
```

Comment:
The Github Markdown renderer doesn't recognize this as a listing (and I believe `readthedocs` will behave the same).

I suspect a blank line is needed above the first point. Possibly the two lines belonging to `2.` need to be indented too, but I'm not sure.